dnf/base.py

```diff
@@ -1612,6 +1730,21 @@ def _install_multiarch(self, query, reponame=None, strict=True):
             self._goal.install(select=sltr, optional=(not strict))
         return len(available)
 
+    def install_module(self, specs):
+        """
+        Install module based on provided specs
+
+        :param specs: group and module specs
+        :return: skipped/not installed group specs or problematic module specs
+        """
+        skipped_specs = specs
+        try:
+            skipped_specs = self.repo_module_dict.install(specs)
+        except NoModuleException:
+            self.repo_module_dict.install(specs[1:])
```

Comment:
This looks strange to me. Can you explain it?

Reply:
I also have a problem to identify the place where NoModuleException could be raised.